apps/product/tasks.py

```
@@ -1,207 +1,983 @@
+                        else:
+                            W_product = False
+                        
+                        yandexmarket_product = product.filter(marketplace_type='yandexmarket')
+                        if yandexmarket_product.exists():
+                                yandexmarket_product = yandexmarket_product.first()
+                        else:
+                            yandexmarket_product = False
+                        
+                        if (not yandexmarket_product) and (not W_product):
+                            product = Product.objects.create(vendor_code=vendor_code, marketplace_type="yandexmarket", barcode=barcode)
+                            product_sale_o = ProductSale.objects.create(product=product,company=company,date=date,warehouse=warehouse,marketplace_type="yandexmarket")
```

Ruff

Local variable `product_sale_o` is assigned to but never used

Remove assignment to unused variable `product_sale_o`

(F841)